x-pack/plugins/lens/public/indexpattern_datasource/operations/layer_helpers.ts

```diff
@@ -612,3 +648,25 @@ function isOperationAllowedAsReference({
     hasValidMetadata
   );
 }
+
+export const DEFAULT_TIME_SCALE = 's' as TimeScaleUnit;
+
+export function adjustTimeScaleLabelSuffix(
+  oldLabel: string,
+  previousTimeScale: TimeScaleUnit | undefined,
+  newTimeScale: TimeScaleUnit | undefined
+) {
+  let cleanedLabel = oldLabel;
+  // remove added suffix if column had a time scale previously
+  if (previousTimeScale) {
+    const suffixPosition = oldLabel.lastIndexOf(` ${unitSuffixesLong[previousTimeScale]}`);
+    if (suffixPosition !== -1) {
+      cleanedLabel = oldLabel.substring(0, suffixPosition);
```

Comment:
I think you should change this implementation before merging, because I think that the reference editor work will cause the time scaling suffixes to get removed accidentally. The reference logic is calling `getDefaultLabel` when modifying columns, so I think the suffix adding needs to move into the operation definitions. A nice benefit of this is that you won't need to do any string manipulation here.

Reply:
Can you clarify a little how you expect it to work? The label change would happen in the getDefaultLabel or buildColumn function of the operations supporting time scaling?